src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int SIZE_ARR = 10;
+    private int size = 0;
+    private final Object[] keyArr = new Object[SIZE_ARR];
+    private final Object[] valueArr = new Object[SIZE_ARR];
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = indexCheck(key);
+
+        if (keyIndex == -1) {
+            keyArr[size] = key;
+            valueArr[size] = value;
+            size++;
+        } else {
+            valueArr[keyIndex] = value;
+        }
```

Comment:
just to improve readability a bit more
```suggestion
            return;
        }
        valueArr[keyIndex] = value;
```